instrumentation/jms-1.1/src/jms2Test/groovy/SpringListenerJms2Test.groovy

```diff
@@ -18,17 +18,21 @@ class SpringListenerJms2Test extends AgentTestRunner {
     def context = new AnnotationConfigApplicationContext(Config)
     def factory = context.getBean(ConnectionFactory)
     def template = new JmsTemplate(factory)
+    // TODO(anuraaga): There is no defined order between when JMS starts receiving and our attempt
```

Comment:
My exporter orders everything by startTimeNanos - the previous one had some strange ordering hacks in it (presumably since things were timed by micros instead of nanos?). I feel like previous successes for JMS were coincidence (possibly helped by gradle-retry-plugin) and the best fix will be to make assertTraces order-agnostic

Reply:
previously the ordering was based on when the startSpan() was _called_, not based on the _start time_, and the JMS receive instrumentation is weird in that it doesn't start the span until the message is received, and back-dates it to when receive call was started, which is why the test ordering flip flopped a bit in the JMS tests (I was wondering this today when reviewing also)